### DISCUSSION TITLE
Can I / should I update gradio inside the container?

### DISCUSSION THREAD

**Discussion by hugalafutro** [AbdBarho/stable-diffusion-webui-docker#391] [Q&A]
Hi,
image browser extension wasn't working right recently and after recent update stopped completely with error it requires newer gradio, I was wondering if I can do this inside the container and if so then how? or if you will later release image with updated version? More info @ https://github.com/AlUlkesh/stable-diffusion-webui-images-browser/issues/116#issuecomment-1493259585

**Comment by AbdBarho** [comment]
you can do what ever you want in [the startup script](https://github.com/AbdBarho/stable-diffusion-webui-docker/wiki/Usage#extensions)
```
pip install -U gradio
```

**Reply by hugalafutro** [reply]
Hi, I have tried that, but now the whole container crashes on start and and I've no idea how to reverse it
```
webui-docker-auto-1  | Traceback (most recent call last):
webui-docker-auto-1  |   File "/stable-diffusion-webui/webui.py", line 337, in <module>
webui-docker-auto-1  |     webui()
webui-docker-auto-1  |   File "/stable-diffusion-webui/webui.py", line 281, in webui
webui-docker-auto-1  |     create_api(app)
webui-docker-auto-1  |   File "/stable-diffusion-webui/webui.py", line 199, in create_api
webui-docker-auto-1  |     from modules.api.api import Api
webui-docker-auto-1  |   File "/stable-diffusion-webui/modules/api/api.py", line 8, in <module>
webui-docker-auto-1  |     from gradio.processing_utils import decode_base64_to_file
webui-docker-auto-1  | ImportError: cannot import name 'decode_base64_to_file' from 'gradio.processing_utils' (/usr/local/lib/python3.10/site-packages/gradio/processing_utils.py)
```

edit: I managed to reverse it by force deleting the container with volumes in lazydocker, but I still would like to be able to update gradio so I can use imagebrowser. I would appreciate any insights, thanks!

edit2: I spoke too soon, now all the settings in webui  are broken I'm going to try to recreate the container from scratch

**Reply by AbdBarho** [reply]
just remove the `pip install -U gradio` line and remove the container, then run again, you should get a fresh container with everything reset, you can also delete `data/config/auto/settings.json` if you want to

**Reply by hugalafutro** [reply]
Thanks! I managed to reset the container like that. I installed webui directly on the host and got gradio: 3.23.0 out of the box and image browser works again, so I'll use that as "main" and keep the container as backup in case I break something again.

edit: omg I should've done that form the get go I'm just used to putting everything in docker. I can now render at 2048x2048 with 3090
